src/OrchardCore/OrchardCore.Users.Core/Extensions/UsersServiceCollectionExtensions.cs

```diff
@@ -65,5 +65,17 @@ public static IServiceCollection AddUsers(this IServiceCollection services)
 
             return services;
         }
+
+        public static IServiceCollection TryAddDefaultEmailTokenProvider(this IServiceCollection services)
+        {
+            var emailTokenProviderType = typeof(EmailTokenProvider<>).MakeGenericType(typeof(IUser));
```

Comment:
please read my comment [here](https://github.com/OrchardCMS/OrchardCore/discussions/15585#discussioncomment-8956352)

Reply:
@lampersky yes this change will not fix that. The intention was not to fix that. just clean up to when the these providers were registered. I'll reply to your comment in that thread.